awx/main/tasks.py

```diff
@@ -1706,6 +1706,9 @@ def build_extra_vars_file(self, job, private_data_dir):
         safe_dict = {}
         if job.job_template and settings.ALLOW_JINJA_IN_EXTRA_VARS == 'template':
             safe_dict = job.job_template.extra_vars_dict
+            wj = job.get_workflow_job()
+            if wj and wj.workflow_job_template:
+                safe_dict.update(wj.workflow_job_template.extra_vars_dict)
```

Comment:
This doesn't seem like it respects the original intent of `ALLOW_JINJA_IN_EXTRA_VARS`

Permissions to add a JT to a WFJT are execute_role to the JT and admin_role to the WFJT. These are being considered as prompts, and that's not incorrect. Permission-wise, they are otherwise treated as prompts. This change opens up a backdoor for execute-level people to template stuff.

Reply:
Yea, now that I tinker with this, you're right, because WFJT `extra_vars` are treated like prompts from an RBAC perspective.